nipype/interfaces/io.py

```diff
@@ -2721,18 +2721,25 @@ def _list_outputs(self):
 
 
 class BIDSDataGrabberInputSpec(DynamicTraitedSpec):
-    base_dir = Directory(exists=True,
-                         desc='Path to BIDS Directory.',
-                         mandatory=True)
-    output_query = traits.Dict(key_trait=Str,
-                               value_trait=traits.Dict,
-                               desc='Queries for outfield outputs')
-    raise_on_empty = traits.Bool(True, usedefault=True,
-                                 desc='Generate exception if list is empty '
-                                 'for a given field')
-    return_type = traits.Enum('file', 'namedtuple', usedefault=True)
-    strict = traits.Bool(desc='Return only BIDS "proper" files (e.g., '
-                              'ignore derivatives/, sourcedata/, etc.)')
+    base_dir = Directory(
+        exists=True,
+        desc='Path to BIDS Directory.',
+        mandatory=True)
+    output_query = traits.Dict(
+        key_trait=Str,
+        value_trait=traits.Dict,
+        desc='Queries for outfield outputs')
+    raise_on_empty = traits.Bool(
+        True, usedefault=True,
+        desc='Generate exception if list is empty for a given field')
+    return_type = traits.Enum(
+        'file', 'namedtuple', usedefault=True)
+    index_derivatives = traits.Bool(
+        False, usedefault=True,
+        desc='Index derivatives/ sub-directory')
```

Comment:
This should be `mandatory=True`, since you don't use an `isdefined()` check on it below. It's possible to pass an `Undefined` to things with defaults to disable the input, unless `mandatory` is set `True`.

Reply:
ok!